sdk/cargo-build-sbf/src/main.rs

```diff
@@ -968,6 +1022,24 @@ fn main() {
     let sbf_sdk: PathBuf = matches.value_of_t_or_exit("sbf_sdk");
     let sbf_out_dir: Option<PathBuf> = matches.value_of_t("sbf_out_dir").ok();
 
+    let sbf_tools_version = if matches.is_present("tools_version") {
+        if matches.value_of("tools_version").unwrap() == "latest" {
+            get_latest_sbf_tools_version().unwrap_or_else(|err| {
+                warn!(
+                    "Can't get the latest version of sbf-tools: {}. Using built-in version {}.",
+                    err, &sbf_tools_version,
+                );
+                sbf_tools_version
+            })
+        } else {
+            validate_sbf_tools_version(
```

Comment:
Looks like this'll hit the network (a GET from github) on every `cargo` invocation?

What if instead `--tools-version` just supported whatever tools were already installed, and we had an external method for managing the install of a different tool versions? It feels a little crammed to do it in here.

Reply:
It's somewhat convenient that cargo-build-sbf manages the installed tools, most of the time completely transparently to the developers.  I expect the option `--tools-version` would be used infrequently only to select/install a toolchain version different from the one currently selected. Once the links are set up the same toolchain is used without the option `--tools-vesion` and GET from githib is not done on cargo invocations. I can also improve the logic so that version is validated by checking the filesystem first. If a toolchain with same or newer version is found in the filesystem, GET from github is not needed.

Reply:
I updated the version validation logic to check the filesystem first, so the network is hit only for a version of tools that is not installed locally, in which case cargo-build-sbf would have to download the binary package from github anyway. Michael, what do you think? Still no good?

Reply:
Thanks, your updates seem fine for now.

I think I'd favour removing the "latest" option entirely, unless you really feel strong about it. It introduces support ambiguity (my latest may not be your latest) and generally I feel it's useful for the user to declare what they expect up front anyway

Reply:
I added an option validator and removed the check for "latest" version. Only explicit version strings vX.Y are valid now.